Index: include/llvm/InitializePasses.h
===================================================================
--- include/llvm/InitializePasses.h	(revision 224330)
+++ include/llvm/InitializePasses.h	(working copy)
@@ -19,6 +19,16 @@
 
 class PassRegistry;
 
+#ifdef TSAR_PROJECT
+
+/// initializeTSAR - Initialize all passes linked into the 
+/// TSAR project.
+void initializeTSAR(PassRegistry &Registry);
+
+void initializePrivateRecognitionPassPass(PassRegistry &Registry);
+
+void initializePrivatePrintPassPass(PassRegistry &Registry);
+
+#endif//TSAR_PROJECT
+
 /// initializeCore - Initialize all passes linked into the
 /// TransformUtils library.
 void initializeCore(PassRegistry&);
Index: include/llvm/LinkAllPasses.h
===================================================================
--- include/llvm/LinkAllPasses.h	(revision 224330)
+++ include/llvm/LinkAllPasses.h	(working copy)
@@ -12,6 +12,10 @@
 //
 //===----------------------------------------------------------------------===//
 
+#ifdef TSAR_PROJECT
+#include "tsar_pass.h"
+#endif//TSAR_PROJECT
+
 #ifndef LLVM_LINKALLPASSES_H
 #define LLVM_LINKALLPASSES_H
 
@@ -47,6 +51,10 @@
       if (std::getenv("bar") != (char*) -1)
         return;
 
+#ifdef TSAR_PROJECT
+      (void) llvm::createPrivateRecognitionPass();
+      (void) llvm::createPrivatePrintPass();
+#endif//TSAR_PROJECT
+
       (void) llvm::createAAEvalPass();
       (void) llvm::createAggressiveDCEPass();
       (void) llvm::createAliasAnalysisCounterPass();
